### DISCUSSION TITLE
[E893] Could not find function 'spacy.Tagger.v2' in function registry 'architectures'

### DISCUSSION THREAD

**Discussion by DiegoIE** [explosion/spaCy#11179] [Help: Coding & Implementations] [answered]
I'm trying to user augmenty package, based on Spacy, but keeps producing the same error, even when I have the latest version installed.

```python-traceback
RegistryError: [E893] Could not find function 'spacy.Tagger.v2' in function registry 'architectures'. If you're using a custom function, make 
sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed 
in your environment.


Available names: spacy-legacy.CharacterEmbed.v1, spacy-legacy.EntityLinker.v1, spacy-legacy.HashEmbedCNN.v1, spacy-
legacy.MaxoutWindowEncoder.v1, spacy-legacy.MishWindowEncoder.v1, spacy-legacy.MultiHashEmbed.v1, spacy-legacy.Tagger.v1, 
spacy-legacy.TextCatBOW.v1, spacy-legacy.TextCatCNN.v1, spacy-legacy.TextCatEnsemble.v1, spacy-legacy.Tok2Vec.v1, spacy-
legacy.TransitionBasedParser.v1, spacy-transformers.Tok2VecTransformer.v1, spacy-transformers.Tok2VecTransformer.v2, spacy-
transformers.Tok2VecTransformer.v3, spacy-transformers.TransformerListener.v1, spacy-transformers.TransformerModel.v1, spacy-
transformers.TransformerModel.v2, spacy-transformers.TransformerModel.v3, spacy.CharacterEmbed.v2, spacy.EntityLinker.v1, 
spacy.HashEmbedCNN.v2, spacy.MaxoutWindowEncoder.v2, spacy.MishWindowEncoder.v2, spacy.MultiHashEmbed.v2, 
spacy.PretrainCharacters.v1, spacy.PretrainVectors.v1, spacy.SpanCategorizer.v1, spacy.Tagger.v1, spacy.TextCatBOW.v2, 
spacy.TextCatCNN.v2, spacy.TextCatEnsemble.v2, spacy.TextCatLowData.v1, spacy.Tok2Vec.v2, spacy.Tok2VecListener.v1, 
spacy.TorchBiLSTMEncoder.v1, spacy.TransitionBasedParser.v2
```

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @DiegoIE , can you provide a short script to reproduce this error?

To sanity-check, [`augmenty`](https://github.com/KennethEnevoldsen/augmenty) requires `spacy<3.3.0,>=3.2.0`, so please ensure that your spaCy version is compatible. Can you also copy-paste the output of  `python -m spacy info`?

**Reply by DiegoIE** [reply]
spaCy version    3.4.0
Location         C:\Python310\lib\site-packages\spacy
Platform         Windows-10-10.0.22000-SP0
Python version   3.10.0
Pipelines        en_core_web_sm (3.4.0)

**Reply by DiegoIE** [reply]
I was only trying to run the page example  (https://spacy.io/universe/project/augmenty), but when I had v. 3.2, I got another error asking for an upgrade.

import spacy
import augmenty

nlp = spacy.load('en_core_web_md')

docs = nlp.pipe(['Augmenty is a great tool for text augmentation'])

ent_dict = {'ORG': [['spaCy'], ['spaCy', 'Universe']]}
entity_augmenter = augmenty.load('ents_replace.v1',
                                 ent_dict = ent_dict, level=1)

for doc in augmenty.docs(docs, augmenter=entity_augmenter, nlp=nlp):
    print(doc)

**Reply by ljvmiranda921** [reply]
Ah ok I see it now. So, **the spaCy version should also be compatible with your model's version.** In your last reply, it seems that you installed spaCy 3.2, but your models were only compatible with 3.4. What you can do is (1) uninstall the 3.4 version of the **spaCy model** and (2) re-download its 3.2 version:

```
pip uninstall en_core_web_md==3.4.0
spacy download en_core_web_md  # if your spaCy version is 3.2, then it should download en_core_web_md==3.2
```

You can then use `spacy info` again to confirm if your spaCy library and model versions are compatible, i.e., they should be both 3.2. Again, in order for `augmenty` to work, you should have the 3.2 versions of the spaCy library and the `en_core_web_md` model. Below's my `spacy info` as an example:

```
============================== Info about spaCy ==============================

spaCy version    3.2.0
... some other info
Pipelines        en_core_web_md (3.2.0)
```
